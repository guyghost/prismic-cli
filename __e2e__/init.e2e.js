const path = require('path');
const fs = require('fs');
const { spawnSync } = require('child_process');
const {
  login,
  changeBase,
  deleteRepo,
  TMP_DIR,
  PRISMIC_BIN,
  rmdir,
  mkdir,
  genRepoName,
} = require('./utils');

describe('prismic init', () => {
  jest.setTimeout(300000);

  const repoName = genRepoName('cli-init-test');
  const dir = path.resolve(TMP_DIR, 'init-test');
  const config = path.resolve(dir, 'prismic-configuration.js');

  const args = [
    '--folder', dir,
    '--domain', repoName,
    '--template', 'NodeJS',
    '--noconfirm',
  ];

  beforeAll(async () => {
    changeBase();
    login();
    await deleteRepo(repoName);
    await mkdir(TMP_DIR, { recursive: true });
  });

  beforeEach(async () => rmdir(dir, { recursive: true }));

  it('should initialise a project from a template and create a new repo', () => {
    const res = spawnSync(PRISMIC_BIN, ['init', ...args, '--new'], { encoding: 'utf8', shell: true });
    expect(fs.existsSync(dir)).toBeTruthy();
    expect(res.status).toBeFalsy();
    expect(fs.existsSync(config)).toBe(true);
  });

  it('should initialise a project from a template with an existing repo', () => {   
    const res = spawnSync(PRISMIC_BIN, ['init', ...args], { encoding: 'utf8', shell: true });
    expect(fs.existsSync(dir)).toBeTruthy();
    expect(res.status).toBeFalsy();
    expect(fs.existsSync(config)).toBe(true);
  });
});
