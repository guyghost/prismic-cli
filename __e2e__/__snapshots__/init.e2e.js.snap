// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prismic init should initialise a project from a template and create a new repo 1`] = `
"CAREFUL, your current base is: https://wroom.io

Let's get to it!
Initialize local project
Running npm install...
Your project is ready, to proceed:

Go to the project folder : cd /Users/marc/Projects/prismic/prismic-cli/__tmp__/init-test
To have hot reload: npm install -g nodemon
Start your project: nodemon app.js
Point your browser to: http://localhost:3000
"
`;

exports[`prismic init should initialise a project from a template with an existing repo 1`] = `
"Let's get to it!
Initialize local project
Running npm install...
Your project is ready, to proceed:

Go to the project folder : cd /Users/marc/Projects/prismic/prismic-cli/__tmp__/init-test
To have hot reload: npm install -g nodemon
Start your project: nodemon app.js
Point your browser to: http://localhost:3000
"
`;
